src/main/java/org/apache/commons/io/monitor/FileAlterationMonitor.java

```diff
@@ -165,8 +165,8 @@ public synchronized void stop(final long stopInterval) throws Exception {
         running = false;
         try {
             thread.join(stopInterval);
+            thread.interrupt();
         } catch (final InterruptedException e) {
-            Thread.currentThread().interrupt();
```

Comment:
Tell me if i'm wrong, because it is kind of obscure.

The monitor embed a thread. But when we call `.stop()` we are calling from another thread (most likely the main thread).
Isn't `Thread.currentThread().interrupt();` going to stop the calling thread? In our case the main thread?

Now, i agree that catching an exception to do nothing sounds dumb. But in our case we want to stop the inner thread. So even if it is interupted while performing the `join#` is kind of a good news. It just stopped earlier than expected.

Am i correct?